docs/OV_Runtime_UG/ov_dynamic_shapes.md

```diff
@@ -0,0 +1,122 @@
+
+@snippet snippets/ov_dynamic_shapes.cpp ov_dynamic_shapes:reshape_undefined
+
+To simplify the code, the examples assume that the model has a single input and single output.
+However, there are no limitations on the number of inputs and outputs to apply dynamic shapes.
+
+### Undefined Dimensions "Out Of the Box"
+
+Dynamic dimensions may exist in the input model without calling reshape.
+Many DL frameworks support undefined dimensions.
+If such a model is converted with Model Optimizer or read directly by Core::read_model, undefined dimensions are preserved.
+Such dimensions automatically treated as dynamic ones.
+So you don't need to call reshape if undefined dimensions are already configured in the original model or IR file.
+
+If the input model has undefined dimensions that you are not going to change during the inference, you can set them to static values, using the same `reshape` method of the model.
+From the API perspective any combination of dynamic and static combination can be configured.
+
+Model Optimizer provides capability to reshape the model during the conversion.
+Use this capability to save time on calling `reshape` method in the end application.
+
+### Dimension Bounds
+
+Besides marking a dimension just dynamic, you can also specify lower and/or upper bounds that define a range of allowed values for the dimension.
+Bounds are coded as arguments for `ov::Dimension`:
+
+@snippet snippets/ov_dynamic_shapes.cpp ov_dynamic_shapes:reshape_bounds
+
+Information about bounds gives opportunity for the inference plugin to apply additional optimizations.
+Using dynamic shapes assumes the plugins apply more loose optimization technique during model compilation
+It may require more resources for model compilation and inference.
+So providing any additional information like bounds can be beneficial.
+For the same reason it is not recommended to leave dimensions as undefined without the real need.
+
+When specifying bounds, the lower bound is not so important as upper bound, because knowing of upper bound allows plugins to more precisely allocate memory for intermediate tensors for inference and use lesser number of tuned kernels for different sizes.
+Precisely speaking benefits of specifying lower or upper bound is HW plugin dependent.
+Depending on the plugin specifying upper bounds can be required.
+<TODO: reference to plugin limitations table>.
+If developer knowns lower and upper bounds for dimension it is recommended to specify them even when plugin can execute model without the bounds.
+
+### Setting Input Tensors
+
+Preparing model with the reshape method was the first step.
+The second step is passing a tensor with an appropriate shape to infer request.
+This is similar to [regular steps](Integrate_with_customer_application_new_API.md), but now we can pass tensors with different shapes for the same executable model and even for the same infer request:
```

Comment:
same executable model => same inference request ?

Reply:
I emphasize that this is for the same executable model, which means the compilation (from the user perspective) is done once because we mentioned `compile_model` before. The same inference request implies indeed the same executable model, but this duplication is made intentionally to make sure that a new user connects all these OV terms together.